External/FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -893,6 +904,33 @@ bool ConstProp::ConstantPropagation(IREmitter *IREmit, const IRListView& Current
         IREmit->ReplaceWithConstant(CodeNode, NewConstant);
         Changed = true;
       }
+      else if (SrcIsConstant) {
+        // If the source is constant in a BFI then it can be converted to an or or bic depending.
+        if (ConstantSrc == 0) {
```

Comment:
The structure here is:

```
if (ConstantSrc == 0) {
  // AND case
} else {
  if (HasConsecutiveBits(ConstantSrc, Op->Width)) {
    // OR case
  }
}
```

I was expecting 

```
if (HasConsecutiveBits(ConstantSrc, Op->Width)) {
  if (ConstantSrc & 1) {
    // OR case
  } else {
    // AND case
  }
}
```

At the moment, I think there's a bug if you try to insert the low 16 bits of 0x10000. 0x10000 is constant, and the low 16 bits are the same (all zero), but 0x10000 != 0, so we end up in the OR case, which doesn't make sense.

Reply:
You're right it does end up being a little weird flipped the consecutive bits check around and generates the constant only once since both sides will actually use the same constant regardless.